### ISSUE TITLE
The `ssh` extension generates invalid paths in ssh config file when multiple host patterns are used

### ISSUE THREAD

**Issue by legigor** [Azure/azure-cli-extensions#4181]
### Extension name (the extension in question)

`ssh`

```
{
  "experimental": false,
  "extensionType": "whl",
  "name": "ssh",
  "path": "/Users/username/.azure/cliextensions/ssh",
  "preview": true,
  "version": "1.0.0"
}
```

### Description of issue (in as much detail as possible)

❗ The issue arrived with version `1.0.0` and was not actual for `0.1.*`

When I have multiple hosts in the getting config expression like this 

```
az ssh config --ip '*.internal 10.31.*' --file ~/.ssh/azure/config
```

It populates the following config

```
Host *.yscan.local 10.31.*
	User username@myorg.io
	CertificateFile /Users/username/.ssh/azure/az_ssh_config/*.internal 10.31.*/id_rsa.pub-aadcert.pub
	IdentityFile /Users/username/.ssh/azure/az_ssh_config/*.internal 10.31.*/id_rsa
```

Where `Host` directive is correct, but when I try to connect to a host I get the following error:

```
ssh ys-admin-app.yscan.local
/Users/username/.ssh/azure/config line 4: garbage at end of line; "10.31.*/id_rsa.pub-aadcert.pub".
/Users/username/.ssh/azure/config line 5: garbage at end of line; "10.31.*/id_rsa".
/Users/username/.ssh/azure/config: terminating, 2 bad configuration options            
```

I tried to double-quote the path in the populated config like this

```
CertificateFile "/Users/username/.ssh/azure/az_ssh_config/*.internal 10.31.*/id_rsa.pub-aadcert.pub"
IdentityFile "/Users/username/.ssh/azure/az_ssh_config/*.internal 10.31.*/id_rsa"
```

and then it worked as expected.

**Comment by yonzhan** [comment]
route to service team

**Comment by jsntcy** [comment]
@vthiebaut10, please help take a look.

**Comment by vthiebaut10** [comment]
@legigor Thank you for bringing this problem to our attention. We will have if fixed on the next release of the extension. Meanwhile, another possible workaround is to use the --keys-destination-folder parameter to save the generated keys on a different folder. For example:
`az ssh config --ip '*.internal 10.31.*' --file ~/.ssh/azure/config --keys-destination-folder ~/.ssh/azure/mykeys`

**Comment by Tea23** [comment]
I am experiencing a possible variant of this bug. A developer's user directory contains a space, so `C:\Users\Firstname Lastname`

When they run `az ssh vm --ip ip.of.vm` they get the following:

`command-line line 0: garbage at end of line; "Lastname\\AppData\\Local\\Temp\\aadsshcert9p1jsvkh\\id_rsa.pub-aadcert.pub".`

As an attempt at a workaround we followed @vthiebaut10's suggested workaround and did

`az ssh config --ip ip.of.vm --file e:\config --keys-destination-folder e:\keys`

The same error is still retuerned.

**Comment by vthiebaut10** [comment]
@Tea23 Yes, that is a variant of the same bug. That will be fixed on the next release. For now, creating a config file with a --keys-destination-folder and using that config to connect to your resource should be working.
What exactly happens when you run:
`az ssh config --ip ip.of.vm --file e:\config --keys-destination-folder e:\keys`
`ssh -F e:\config ip.of.vm`

**Comment by vthiebaut10** [comment]
@Tea23 @legigor Both your issues have been addressed in the new version 1.0.1 of the SSH extension. Please validate that it works for you.

**Comment by maertendMSFT** [comment]
closing as it is called out as fixed.
